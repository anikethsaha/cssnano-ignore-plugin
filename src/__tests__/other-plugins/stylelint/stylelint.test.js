import fs from 'fs';

import path from 'path';
import postcss from 'postcss';
import stylelint from 'stylelint';

import cssnanoIgnoreRemove from '../../../remove';
import cssnanoIgnoreAdd from '../../../add';

const run = (fixturename) => {
  const code = fs.readFileSync(
    path.join(__dirname, `/fixtures/${fixturename}`),
    'utf8'
  );
  it(
    'should ignore the cssnano-ignore-line comments for stylelint ' +
      fixturename,
    (done) => {
      postcss([
        cssnanoIgnoreRemove(),
        stylelint({
          config: {
            rules: { 'property-no-unknown': true, 'unit-no-unknown': true },
          },
        }),
        cssnanoIgnoreAdd(),
      ])
        .process(code)
        .then((resultObject) => {
          const stylelintWarnings = resultObject.messages.slice(1);
          expect(stylelintWarnings).toMatchSnapshot();
          done();
        });
    }
  );
};
fs.readdirSync(path.join(__dirname, `./fixtures`)).forEach(run);
