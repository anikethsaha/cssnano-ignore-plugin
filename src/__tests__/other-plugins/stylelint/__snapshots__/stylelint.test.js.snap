// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should ignore the cssnano-ignore-line comments for stylelint property-no-unknown.css 1`] = `
Array [
  Warning {
    "column": 3,
    "line": 2,
    "node": Object {
      "prop": "colr",
      "raws": Object {
        "before": "
  ",
        "between": ": ",
      },
      "source": Object {
        "end": Object {
          "column": 13,
          "line": 2,
        },
        "input": Input {
          "css": "a {
  colr: blue;
}
a {
  /* postcss-ignore-line */
  my-property: 1;
}

a {
  color: green;
}
a {
  fill: black;
}
a {
  -moz-align-self: center;
}
a {
  -webkit-align-self: center;
}
a {
  align-self: center;
}
",
          "hasBOM": false,
          "id": "<input css 1>",
        },
        "start": Object {
          "column": 3,
          "line": 2,
        },
      },
      "type": "decl",
      "value": "blue",
    },
    "plugin": "stylelint",
    "rule": "property-no-unknown",
    "severity": "error",
    "text": "Unexpected unknown property \\"colr\\" (property-no-unknown)",
    "type": "warning",
  },
]
`;

exports[`should ignore the cssnano-ignore-line comments for stylelint unit-no-unknown.css 1`] = `
Array [
  Warning {
    "column": 10,
    "index": 7,
    "line": 2,
    "node": Object {
      "prop": "width",
      "raws": Object {
        "before": "
  ",
        "between": ": ",
      },
      "source": Object {
        "end": Object {
          "column": 18,
          "line": 2,
        },
        "input": Input {
          "css": "a {
  width: 10pixels;
}
a {
  /* postcss-ignore-line */
  width: calc(10px + 10pixels);
}

p {
  height: 10pixels;
}

a {
  width: 10px;
}
a {
  width: 10px;
}
a {
  width: 10px;
}
a {
  width: calc(10px + 10px);
}

/* postcss-ignore */
html {
  height: 100ppxx;
}
",
          "hasBOM": false,
          "id": "<input css 2>",
        },
        "start": Object {
          "column": 3,
          "line": 2,
        },
      },
      "type": "decl",
      "value": "10pixels",
    },
    "plugin": "stylelint",
    "rule": "unit-no-unknown",
    "severity": "error",
    "text": "Unexpected unknown unit \\"pixels\\" (unit-no-unknown)",
    "type": "warning",
  },
  Warning {
    "column": 11,
    "index": 8,
    "line": 10,
    "node": Object {
      "prop": "height",
      "raws": Object {
        "before": "
  ",
        "between": ": ",
      },
      "source": Object {
        "end": Object {
          "column": 19,
          "line": 10,
        },
        "input": Input {
          "css": "a {
  width: 10pixels;
}
a {
  /* postcss-ignore-line */
  width: calc(10px + 10pixels);
}

p {
  height: 10pixels;
}

a {
  width: 10px;
}
a {
  width: 10px;
}
a {
  width: 10px;
}
a {
  width: calc(10px + 10px);
}

/* postcss-ignore */
html {
  height: 100ppxx;
}
",
          "hasBOM": false,
          "id": "<input css 2>",
        },
        "start": Object {
          "column": 3,
          "line": 10,
        },
      },
      "type": "decl",
      "value": "10pixels",
    },
    "plugin": "stylelint",
    "rule": "unit-no-unknown",
    "severity": "error",
    "text": "Unexpected unknown unit \\"pixels\\" (unit-no-unknown)",
    "type": "warning",
  },
]
`;
